app/infrastructure/cache/redis_cache.py

```
@@ -0,0 +1,230 @@
+            Number of keys deleted.
+        """
+        if not self.is_available() or self.redis is None:
+            return 0
+        try:
+            deleted = 0
+            async for k in self.redis.scan_iter(match=pattern):
+                await self.redis.delete(k)
+                deleted += 1
+            if deleted > 0:
+                logger.info("Cache INVALIDATE: %s (%s keys)", pattern, deleted)
+            return deleted
```

Ruff (0.14.14)

[warning] Consider moving this statement to an `else` block

(TRY300)